qemu/tests/qemu_guest_agent.py

```diff
@@ -1828,6 +1828,27 @@ def gagent_check_nonexistent_cmd(self, test, params, env):
         else:
             test.fail("Should return error info.")
 
+    @error_context.context_aware
+    def gagent_check_with_migrate(self, test, params, env):
+        """
+        Migration test with guest agent service running.
+
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+        error_context.context("Migrate guest while guest agent service is"
+                              " running.", logging.info)
+        self.vm.monitor.migrate_set_speed(params.get("mig_speed", "1G"))
+        self.vm.migrate()
+        error_context.context("Recreate a QemuAgent object after vm"
+                              " migration.", logging.info)
+        self.gagent = False
```

Comment:
`self.agent` is not a bool,  better use `None`.

Reply:
@yanan-fu agree, updated.